### PR TITLE
Use go_run instead of run_go_tool

### PR DESCRIPTION
**Pull request by chmouel**
# Changes

:broom: As per the error message whhen running update-codegen.sh

```console
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! !!!! WARN: The "run_go_tool" function is deprecated. Use "go_run" instead. !!!! !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```
run_go_tool function has been deprecated in favour of go_run

/kind misc

Fixes #

**Release Note**

```release-note

```

**Docs**

```docs

```

### CODE REVIEW

**Review by cardil — APPROVED** [review]

**Comment by dprotaso** [comment]
@cardil have you see this?

> Error: stat eval-all: no such file or directory
> exit status 1

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -22,7 +22,7 @@ source $(dirname $0)/../vendor/knative.dev/hack/codegen-library.sh
 export PATH="$GOBIN:$PATH"
 
 function run_yq() {
-  run_go_tool github.com/mikefarah/yq/v4@v4.23.1 yq "$@"
+	go_run github.com/mikefarah/yq/v4@v4.23.1 "$@"
```
Comment (dprotaso) [inline hack/update-codegen.sh:25]: note: we had to drop the extra `yq` argument to the invocation

**Pull request merged**